docs/eventing/sources/containersource/README.md

```diff
@@ -0,0 +1,224 @@
+# Creating a ContainerSource object
+
+![API version v1](https://img.shields.io/badge/API_Version-v1-red?style=flat-square)
+
+This topic describes how to configure ContainerSource as an event source for
+functions.
+
+The ContainerSource object starts a container image that generates events and
+sends messages to a sink URI. You can also use ContainerSource to support your
+own event sources in Knative.
+
+In the examples below, the event source is a heartbeats container and the sink
+is a Knative Service.
+If you have an existing event source and sink, you can replace the examples with
+your own values.
+
+## Before you begin
+
+Before you can create a ContainerSource object:
+
+- You must have [Knative Eventing](../../../admin/install/install-eventing-with-yaml)
+installed on your cluster.
+- If you want to use the example heartbeats event source below, you must also:
+    - Install [ko](https://github.com/google/ko)
+    - Set `KO_DOCKER_REPO`. For example, `gcr.io/[gcloud-project]` or `docker.io/<username>`
+    - Authenticate with your `KO_DOCKER_REPO`
+    - Install [`docker`](https://docs.docker.com/install/)
+
+## Create a ContainerSource object
+
+1. Build an image of your event source and publish it to your image repository.
+Your image must read the environment variable `K_SINK` and post messages to the
+URL specified in  `K_SINK`. If you do not already have an image, you can use
+the following example heartbeats event source by running the commands:
+
+    ```bash
+    git clone -b "{{ branch }}" https://github.com/knative/eventing.git
+    ```
+
+    ```bash
+    ko publish ko://knative.dev/eventing/cmd/heartbeats
+    ```
+
+1. Create a namespace for your ContainerSource by running the command:
+
+    ```bash
+    kubectl create namespace <namespace>
+    ```
+
+    Where `<namespace>` is the namespace that you want your ContainerSource to use.
+    For example, `containersource-example`.
+
+1. Create a sink. If you do not already have a sink, you can use the following Knative
+Service, which dumps incoming messages into its log, by running the command:
+
+    !!! note
+        To create a Knative service you must have Knative Serving installed on your cluster.
+
+    === "kn"
+
+        ```bash
+        kn service create event-display --port 8080 --image gcr.io/knative-releases/knative.dev/eventing/cmd/event_display
+        ```
+
+    === "YAML"
+
+        ```yaml
+        kubectl -n containersource-example apply -f - << EOF
```

Comment:
same as below, for all of `<< EOF` I think it should be `<<EOF` - if not, our style guide needs an update https://dev-knative.netlify.app/development/help/contributor/style-guide/documenting-code/#documenting-yaml

idk if this actually matters but just thought I'd point it out in case, not sure if you're testing these commands